lib/net.c

```diff
@@ -54,6 +52,23 @@ uint8_t default_rss_key[GATEKEEPER_RSS_KEY_LEN] = {
 	0x6a, 0x42, 0xb7, 0x3b, 0xbe, 0xac, 0x01, 0xfa,
 };
 
+
+/* Function to randomize the default RSS key */
+void randomize_rss_key(){
+	srand(time(NULL));
+
+	/* key_initiator gets a random value each time which is xor'd with other values */
+	uint8_t key_intiator = rand();
+	default_rss_key[0] = default_rss_key[0]^key_intiator;
+
+	for(int i=1; i<GATEKEEPER_RSS_KEY_LEN; i++)
+	{
+		default_rss_key[i] = default_rss_key[i]^default_rss_key[i-1];
+	} 
+}
```

Comment:
There's no test ensuring that the new RSS key is an acceptable key. For example, a key with all bits zero won't work.